moodle-scalable-cluster-ubuntu/azuredeploy.json

```diff
@@ -394,7 +476,7 @@
                     }
                 },
                 "templateLink": {
-                    "uri": "[concat(variables('moodleCommon').baseTemplateUrl,'redis.json',parameters('_artifactsLocationSasToken'))]"
+                    "uri": "[concat(variables('moodleCommon').baseTemplateUrl,'redis', if(parameters('redisDeploySwitch'), '', '-none'), '.json',parameters('_artifactsLocationSasToken'))]"
```

Comment:
any reason we can't use the condition property for this?

Reply:
That's exactly what I tried at the beginning, and it kept failing, leading me to conclude that the ARM's `if(,,)` function is eager and causes this issue. Let me explain further in detail.

Basically, I wanted something like the following for the `redisKey` value in `controller.json`:
```
if([condition],listKeys([redisResourceId]).primaryKey),'None')
```
The `[condition]` would be true iff the `redisDeploySwitch` parameter is `true`. However, even when the switch is false, ARM keeps evaluating the `listKeys([redisResourceId]).primaryKey)` and complaining that there's no such resource (because no redis will be deployed when the switch is false). This was unexpected to me (as I was expecting a programming-language-like behavior with `if(,,)`), but then I realized that the parameter evaluations in `if(,,)` might be eager, call-by-value, and all 3 parameters are evaluated even if the condition is false, which is clearly different from the usual programming-language-like behavior of `if ... else ...` construct.

I've been wondering if this understanding of mine is correct, and who to ask about this. Could you confirm if this is the currently expected behavior? If not, could you let me know who to ask about this?

Bottom line is that I did/do want to get rid of this hack, but the ARM's `if(,,)` semantics I experienced doesn't allow that, so I couldn't avoid this hack. Please advise.

Reply:
Thanks. It's working now as desired and fixed in the newly added commit.